test/e2e/serial/workload_placement_test.go

```diff
@@ -151,3 +165,94 @@ func filterNodesWithMoreThanOneNUMAZone(nodes []corev1.Node, nodeResourceTopolog
 
 	return ret, nil
 }
+
+func getWorkloadRequirements(nodeResourceTopologies []nrtv1alpha1.NodeResourceTopology, nodes []corev1.Node) (corev1.ResourceList, error) {
+	for _, nrt := range nodeResourceTopologies {
+		// for all the zones
+		for _, zone := range nrt.Zones {
+			// find zones bigger than the current one
+			zonesMap, err := zonesBiggerThan(zone, nodeResourceTopologies)
+			if err != nil {
+				return nil, err
+			}
+			if len(zonesMap) == 0 {
+				// if no other zone matches but the zone itself.
+				// find if any node but the one this zone comes from can match the requests
+				for _, node := range nodes {
+					if node.Name == nrt.Name {
+						continue
+					}
+					//convert zone resources into a resource request
+					zoneResources := resourceInfoListToResourceList(&zone.Resources)
+					if nodeAllocatableResourcesMatchesRequest(node, zoneResources) {
+						// if there is another node then this is our request
+						// because any pod with this resource request could be sheduled
+						// both to the node or to the NUMA zone, so it is ok for our testing purposes
+						return zoneResources, nil
+					}
+				}
+			}
+		}
+	}
+	return nil, fmt.Errorf("not suitable requirement found")
+}
+
+func resourceInfoListToResourceList(origin *nrtv1alpha1.ResourceInfoList) corev1.ResourceList {
+	ret := make(corev1.ResourceList)
+	for _, resInfo := range *origin {
+		ret[corev1.ResourceName(resInfo.Name)] = resInfo.Available
+	}
+	return ret
+}
+
+func nodeAllocatableResourcesMatchesRequest(node corev1.Node, request corev1.ResourceList) bool {
+
+	for reqName, reqQuantity := range request {
+		nodeQuantity, found := node.Status.Allocatable[reqName]
+		if !found {
```

Comment:
this is worth an error, because it should never happen

Reply:
Why it should never happen? If we have nodes with heterogeneous resources we could have one node with less amount of resources than a numa zone in another node

Reply:
right, in your flow you are considering node candidates, so this can happen indeed.